### ISSUE TITLE
Allow setting variables for fixtures in a more generic way

### ISSUE THREAD

**Issue by ekouts** [reframe-hpc/reframe#2757]
The [docs](https://reframe-hpc.readthedocs.io/en/stable/manpage.html#cmdoption-S) say:

> -S, --setvar=[TEST.]VAR=VAL
    Set variable VAR in all tests or optionally only in test TEST to VAL.
    TEST can have the form [TEST.][FIXT.]*, in which case VAR will be set in fixture FIXT of TEST. Note that this syntax is recursive on fixtures, so that a variable can be set in a fixture arbitrarily deep. TEST prefix refers to the test class name, not the test name, but FIXT refers to the fixture name inside the referenced test.

As far as I can tell `[TEST.][FIXT.]*` is not correct. We can set a variable in all the non-fixture tests with `-Svar=val` but for fixtures we have to do it manually since `FIXT` is not the class name but the name inside a test. Also `*` doesn't work for me at all, not sure what a valid example would be.

I think we should extend `-S` and allow setting fixture variables more easily.

This is also slightly related to  #2626 , since `-S strict_check=1` is not really equivalent to `--strict`.

**Comment by vkarak** [comment]
There are two parts in this issue, I think:

1. The fact that `-S` does not try to set variables in fixtures.
2. The fact that the `.` operator cannot be applied recursively in fixtures despite that being documented.

I think that the first is a feature request whereas the second seems like a bug to me. Perhaps we should separate them?

**Comment by ekouts** [comment]
Sure, I will leave this as new feautre and open a new with the tag `bug` and `documentation`.

**Comment by vkarak** [comment]
This was actually fixed by #2746. Variables are set also in fixtures here:

https://github.com/reframe-hpc/reframe/blob/fb385e10dbe8b6fc47c440bc004ffe5b50c06d4e/reframe/core/decorators.py#L134-L135

This applies also with the `-S` option. For example, trying this sets the environment variables also in the build fixture:

```console
./bin/reframe -C tutorials/cscs-webinar-2022/config/mysettings.py -c tutorials/cscs-webinar-2022/tests/stream7.py -S env_vars=FOO:1 --dry-run
cat stage/tresa/default/gnu/stream_build_bed1cb01/rfm_build.sh
```

The only remaining thing is to issue warnings for unset variables also for fixtures.
